### PR TITLE
[ADD] ansible_base: Add core.role and other submodules

### PR DESCRIPTION
**Pull request by tedsalmon**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  # Create ansible.cfg with correct roles_path
+  - printf '[defaults]\nroles_path=../' >ansible.cfg
+
+script:
+  # Basic role syntax check
+  - ansible-playbook roles/core.role/tests/test.yml -i  roles/core.role/tests/inventory --syntax-check
+
+# Not listed in Galaxy
+#notifications:
+#  webhooks: https://galaxy.ansible.com/api/v1/notifications/
```
Comment (lasley) [inline .travis.yml]: Remove commented code

File: README.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+Ansible Base System Playbook
+============================
+
+This playbook should be used to provision basic roles on a newly deployed server.
+You should make sure to init all submodules, like so:
+```
+git submodule update --init --recursive
+```
+
```
Comment (lasley) [inline README.md]: or `git clone --recursive $URI`

File: README.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+See the individual roles for the configuration options that are available. Use
+ settings.yml and settings.vault.yml for your custom options.
+
+Usage
+=====
+
+Use like any other playbook.
+
+Variables
```
Comment (lasley) [inline README.md]: Remove empty headers

File: roles/core.role/defaults/main.yml [hunk 4]
```diff
@@ -1,2 +1,27 @@
 ---
 # defaults file for core
+
+# System-wide hostname
+system_hostname: "ec2-xxxx-prod-vml-01.example.com"
```
Comment (lasley) [inline roles/core.role/defaults/main.yml]: Does this get applied to all servers in the playbook?

File: roles/core.role/defaults/main.yml [hunk 5]
```diff
@@ -1,2 +1,27 @@
...
+
+# "ignoreip" can be an IP address, a CIDR mask or a DNS host
+fail2ban_config_ignoreip: 127.0.0.1/8
+fail2ban_config_bantime: 600
+fail2ban_config_maxretry: 4
+fail2ban_config_destemail: root@localhost
+
+# Jails
+fail2ban_config_jail_ssh_enabled: yes
+fail2ban_config_jail_sshddos_enabled: yes
```
Comment (lasley) [inline roles/core.role/defaults/main.yml]: newline EOL

Comment (lasley) [reply roles/core.role/defaults/main.yml]: https://stackoverflow.com/questions/36043061/automatically-add-newline-on-save-in-pycharm

File: roles/core.role/meta/main.yml [hunk 6]
```diff
@@ -1,222 +1,26 @@
...
-    # List tags for your role here, one per line. A tag is
-    # a keyword that describes and categorizes the role.
-    # Users find roles by searching for tags. Be sure to
-    # remove the '[]' above if you add tags to this list.
-    #
-    # NOTE: A tag is limited to a single word comprised of
-    # alphanumeric characters. Maximum 20 tags per role.
-
+  author: Ted Salmon
+  description: Role allows you to provision a new Fedora 19+ Server
```
Comment (lasley) [inline roles/core.role/meta/main.yml]: Fedora? I thought we were going to use CentOS?

File: roles/core.role/tasks/apps.yml [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    - nano
+    - python
+    - python-devel
+    - python-pip
+    - firewalld
+    - bzip2
+
+- name: Ensure that firewalld is enabled
+  service: name=firewalld state=started enabled=yes
+
```
Comment (lasley) [inline roles/core.role/tasks/apps.yml]: too many newlines

File: roles/core.role/tasks/fail2ban.yml [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+# fail2ban tasks for core role
+
+- name: Install fail2ban on the system
+  yum: name=fail2ban state={{ fail2ban_pkg_state }}
+  tags: package
+
+- name: configure fail2ban local jails
+  action: template src=jail.local.j2 dest=/etc/fail2ban/jail.local mode=0644
+  notify: restart-fail2ban
+  tags: configuration
```
Comment (lasley) [inline roles/core.role/tasks/fail2ban.yml]: newline eol

File: roles/core.role/tasks/apps.yml [hunk 9]
```diff
@@ -0,0 +1,23 @@
+---
```
Comment (lasley) [inline roles/core.role/tasks/apps.yml:4]: header

File: roles/core.role/meta/main.yml [hunk 10]
```diff
@@ -1,222 +1,26 @@
+# Copyright 2016 LasLabs Inc.
```
Comment (lasley) [inline roles/core.role/meta/main.yml]: 2017

File: roles/core.role/defaults/main.yml [hunk 11]
```diff
@@ -1,2 +1,27 @@
 ---
```
Comment (lasley) [inline roles/core.role/defaults/main.yml:4]: header

File: roles/core.role/tasks/main.yml [hunk 12]
```diff
@@ -1,2 +1,11 @@
...
+# License Apache 2.0 (https://www.apache.org/licenses/LICENSE-2.0.html).
+
 ---
 # tasks file for core
+
+- include: apps.yml tags=applications
+- include: fail2ban.yml tags=fail2ban
+- include: preferences.yml tags=system-preferences
+- include: sshd.yml tags=openssh
+
```
Comment (lasley) [inline roles/core.role/tasks/main.yml:11]: newline

File: roles/core.role/tasks/preferences.yml [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+
+- name: Set the system hostname
+  shell: hostnamectl set-hostname {{ system_hostname }}
+
+- name: Set prefered timezone
+  file: src=/usr/share/zoneinfo/{{ system_timezone }}
+        dest=/etc/localtime state=link force=yes
+
+- name: Verify Root has a .ssh directory
+  file: path=/root/.ssh state=directory
```
Comment (lasley) [inline roles/core.role/tasks/preferences.yml]: nleol

File: roles/core.role/tasks/sshd.yml [hunk 14]
```diff
@@ -0,0 +1,9 @@
+# Copyright 2017 LasLabs Inc.
+# License Apache 2.0 (https://www.apache.org/licenses/LICENSE-2.0.html).
+
+---
+# SSH Server Hardening in core role
+
+- name: Copy in SSH Server Config
+  template: src=sshd_config.j2 dest=/etc/ssh/sshd_config
+  notify: restart-sshd
```
Comment (lasley) [inline roles/core.role/tasks/sshd.yml]: nleol

File: roles/core.role/templates/jail.local.j2 [hunk 15]
```diff
@@ -0,0 +1,31 @@
...
+filter   = sshd
+logpath  = /var/log/secure
+
+
+[ssh-ddos]
+
+enabled  = {{ fail2ban_config_jail_sshddos_enabled }}
+port     = ssh
+filter   = sshd-ddos
+logpath  = /var/log/secure
```
Comment (lasley) [inline roles/core.role/templates/jail.local.j2]: nleol

File: roles/core.role/tests/test.yml [hunk 16]
```diff
@@ -2,4 +2,4 @@
 - hosts: localhost
   remote_user: root
   roles:
-    - core
\ No newline at end of file
+    - core.role
```
Comment (lasley) [inline roles/core.role/tests/test.yml]: nleol

File: roles/core.role/tests/test.yml [hunk 17]
```diff
@@ -2,4 +2,4 @@
 - hosts: localhost
```
Comment (lasley) [inline roles/core.role/tests/test.yml:6]: header

File: vars/settings.vault.yml [hunk 18]
```diff
@@ -0,0 +1,2 @@
+# Copyright 2017 LasLabs Inc.
+# License Apache 2.0 (https://www.apache.org/licenses/LICENSE-2.0.html).
```
Comment (lasley) [inline vars/settings.vault.yml]: nleol

File: vars/settings.yml [hunk 19]
```diff
@@ -0,0 +1,2 @@
+# Copyright 2017 LasLabs Inc.
+# License Apache 2.0 (https://www.apache.org/licenses/LICENSE-2.0.html).
```
Comment (lasley) [inline vars/settings.yml]: nleol

**Review by lasley — CHANGES_REQUESTED** [review]

File: roles/core.role/meta/main.yml (hunk 6 above)
Comment (tedsalmon) [reply roles/core.role/meta/main.yml]: CentOS is not a valid distro in Ansible terms. CentOS 7 is Fedora 19+ ;)

File: roles/core.role/defaults/main.yml (hunk 4 above)
Comment (tedsalmon) [reply roles/core.role/defaults/main.yml]: I added a section about this in the README and removed the hostname default. It should be set in the inventory, per host.

**Comment by tedsalmon** [comment]
Alright @lasley - Tests fixed and comments attended to :)

**Review by lasley — APPROVED** [review]

**Pull request merged**